### PR TITLE
Generate debugging launch.json for all openpilot processes

### PR DESCRIPTION
**Pull request by jakethesnake420**
Run `tools/vscode_debug/generate_vscode_debug_launch.py` to automatically populate the launch.json file with all the openpilot processes. It also supports default arguments for debugging and environment variables.
Generating does not delete your existing launch.json, it will rename it.

image
image
image

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
I don't use VSCode but this seems useful for those who do. However, this is way too many lines, and seems to require the user knows about this and runs it?

Is there a way for VSCode to generate these automatically if you open the project?

File: tools/vscode_debug/generate_vscode_debug_launch.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      "id": proc_name,
+      "type": "promptString",
+      "description": f"Enter the arguments for {proc_name}:",
+      "default": args
+    }
+    for proc_name, args in default_args.items()
+  ]
+
+  # Add a generic input for processes without specific defaults
+  inputs.append({
```
Comment (jnewb1) [inline tools/vscode_debug/generate_vscode_debug_launch.py:114]: there are dropdown options in vscode, can we instead just have 1 config for cpp/python that you can select which process to debug?

**Comment by jnewb1** [comment]
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "inputs": [
        {
          "id": "python_process",
          "type": "pickString",
          "description": "Select the process to debug",
          "options": [
            "selfdrive/controls/controlsd.py",
            "system/timed/timed.py"
          ],
        },
        {
          "id": "args",
          "description": "Arguments to pass to the process",
          "type": "promptString"
        }
    ],
    "configurations": [
        {
            "name": "Python: Openpilot Process",
            "type": "debugpy",
            "request": "launch",
            "program": "${input:python_process}",
            "console": "integratedTerminal",
            "justMyCode": true,
            "args": "${input:args}"
        }
    ]
}
```

something like this would be simpler

**Comment by jakethesnake420** [comment]
> I don't use VSCode but this seems useful for those who do.

I'm curious. It's there a standard ide setup that comma employees use?

**Comment by jnewb1** [comment]
> I'm curious. It's there a standard ide setup that comma employees use?

no, everyone can use their own preferred ide

**Comment by jnewb1** [comment]
I think trying to copy the setup from the main vscode repo  here would make sense. They have the vscode folder checked in with launch configs that work with their  [devcontainer](https://github.com/microsoft/vscode/tree/main/.devcontainer). If we can do something like it suggested above in this comment where we have 1 config for cpp and 1 for python that would be ideal. https://github.com/commaai/openpilot/pull/31197#issuecomment-1924539753.

**Pull request closed without merging**